### ISSUE TITLE
check that data blocks (`script` elements with non-JS type) must be embedded inline?

### ISSUE THREAD

**Issue by rdeltour** [validator/validator#772]
A `script` element with a `type` attribute set to neither a JavaScript MIME type essence match or the string "`module`" represents a [data block](https://html.spec.whatwg.org/#data-block).

The spec also says:
> When used to include data blocks, the data must be embedded inline, the format of the data must be given using the type attribute, and the contents of the script element must conform to the requirements defined for the format used. The `src`, `async`, `nomodule`, `defer`, `crossorigin`, `integrity`, and `referrerpolicy` attributes must not be specified.

I would expect the following document to raise a warning, but it is currently checked with no errors or warnings (using https://validator.w3.org/nu)

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Title</title>
    <script src="https://example.org/#data" type="application/vnd.data"></script>
</head>
<body>
    <h1>Test</h1>
</body>
</html>
```

**Comment by sideshowbarker** [comment]
Yeah, agreed, this is bug in the checker. It should report an error for this case — because the spec requirement says the `src` attribute “must not” be specified (if the `type` value is a non-JavaScript media type).
